### PR TITLE
[DRWTSN32] Implement arm context reading

### PR DESCRIPTION
**Pull request by learn-more**
Please note that this is not verified yet, but it builds (tm)

### CODE REVIEW

**Review by binarymaster — APPROVED** [review]
Ticket id [CORE-17605](https://jira.reactos.org/browse/CORE-17605) :slightly_smiling_face: Otherwise LGTM

File: base/applications/drwtsn32/main.cpp [hunk 1]
```diff
@@ -89,6 +97,16 @@ static void PrintThread(FILE* output, DumpData& data, DWORD tid, ThreadData& thr
 #if defined(_M_IX86) || defined(_M_AMD64)
         xfprintf(output, "dr0:%p dr1:%p dr2:%p dr3:%p dr6:%p dr7:%p" NEWLINE,
                  ctx.Dr0, ctx.Dr1, ctx.Dr2, ctx.Dr3, ctx.Dr6, ctx.Dr7);
+#elif defined(_M_ARM)
+        for (int n = 0; n < ARM_MAX_BREAKPOINTS; ++n)
+            xfprintf(output, "Bvr%d:%p%s", n, ctx.Bvr[n], ((n + 1) == ARM_MAX_BREAKPOINTS) ? NEWLINE : " ");
```
Comment (learn-more) [inline base/applications/drwtsn32/main.cpp:102]: Windbg only shows `bvr[0]` and `bvr[1]`

File: base/applications/drwtsn32/main.cpp [hunk 2]
```diff
@@ -89,6 +97,16 @@ static void PrintThread(FILE* output, DumpData& data, DWORD tid, ThreadData& thr
...
+        for (int n = 0; n < ARM_MAX_BREAKPOINTS; ++n)
+            xfprintf(output, "Bcr%d:%p%s", n, ctx.Bcr[n], ((n + 1) == ARM_MAX_BREAKPOINTS) ? NEWLINE : " ");
```
Comment (learn-more) [inline base/applications/drwtsn32/main.cpp:104]: Windbg only shows `Bcr[0]` and `Bcr[1]`

Comment (learn-more) [reply base/applications/drwtsn32/main.cpp:104]: image

File: base/applications/drwtsn32/stacktrace.cpp [hunk 3]
```diff
@@ -38,25 +38,37 @@ void PrintStackBacktrace(FILE* output, DumpData& data, ThreadData& thread)
...
+    StackFrame.AddrPC.Offset = thread.Context.Pc;
+    StackFrame.AddrStack.Offset = thread.Context.Sp;
+    StackFrame.AddrFrame.Offset = thread.Context.R11;
+#elif defined(_M_ARM64)
+    MachineType = IMAGE_FILE_MACHINE_ARM64;
+    StackFrame.AddrPC.Offset = thread.Context.Pc;
+    StackFrame.AddrStack.Offset = thread.Context.Sp;
+    StackFrame.AddrFrame.Offset = thread.Context.u.s.Fp;
+#else
+#error "Unknown architecture"
```
Comment (SergeGautherie) [inline base/applications/drwtsn32/stacktrace.cpp:69]: The other places do not use quotes...

File: base/applications/drwtsn32/main.cpp [hunk 4]
```diff
@@ -62,10 +62,15 @@ static void PrintThread(FILE* output, DumpData& data, DWORD tid, ThreadData& thr
         xfprintf(output, "eax:%p ebx:%p ecx:%p edx:%p esi:%p edi:%p" NEWLINE,
                  ctx.Eax, ctx.Ebx, ctx.Ecx, ctx.Edx, ctx.Esi, ctx.Edi);
 #elif defined(_M_AMD64)
-        xfprintf(output, "rax:%p rbx:%p rcx:%p rdx:%p rsi:%p rdi:%p" NEWLINE,
-                 ctx.Rax, ctx.Rbx, ctx.Rcx, ctx.Rdx, ctx.Rsi, ctx.Rdi);
+        xfprintf(output, "rax:%p rbx:%p rcx:%p rdx:%p rsi:%p rdi:%p rbp:%p rsp:%p" NEWLINE,
+                 ctx.Rax, ctx.Rbx, ctx.Rcx, ctx.Rdx, ctx.Rsi, ctx.Rdi, ctx.Rbp, ctx.Rsp);
         xfprintf(output, "r8:%p r9:%p r10:%p r11:%p r12:%p r13:%p r14:%p r15:%p" NEWLINE,
                  ctx.R8, ctx.R9, ctx.R10, ctx.R11, ctx.R12, ctx.R13, ctx.R14, ctx.R15);
+#elif defined(_M_ARM)
```
Comment (SergeGautherie) [inline base/applications/drwtsn32/main.cpp:69]: No `defined(_M_ARM64)` in this file for the time being?

**Comment by learn-more** [comment]
Cannot test due to https://jira.reactos.org/browse/CORE-17614

**Comment by binarymaster** [comment]
> Cannot test due to [CORE-17614](https://jira.reactos.org/browse/CORE-17614)

Please retry with my patch attached to that ticket. :slightly_smiling_face:

**Review by binarymaster — APPROVED** [review]
Since CORE-17614 is fixed now, I retested this PR on top of master branch, and it works! :+1: 

Posted the picture and appcrash log in the chat.

File: base/applications/drwtsn32/main.cpp [hunk 5]
```diff
@@ -79,6 +84,9 @@ static void PrintThread(FILE* output, DumpData& data, DWORD tid, ThreadData& thr
 #elif defined(_M_AMD64)
         xfprintf(output, "rip:%p rsp:%p rbp:%p" NEWLINE,
                  ctx.Rip, ctx.Rsp, ctx.Rbp);
+#elif defined(_M_ARM)
+        xfprintf(output, " sp:%p lr:%p pc:%p cpsr:%p" NEWLINE,
```
Comment (binarymaster) [inline base/applications/drwtsn32/main.cpp:88]:
```suggestion
        xfprintf(output, "sp:%p lr:%p pc:%p cpsr:%p" NEWLINE,
```

**Pull request merged**